app/main.py

```diff
@@ -1,3 +1,80 @@
+from __future__ import annotations
+from typing import Union
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(
+            self,
+            km: int | float
+    ) -> None:
+        self.km = km
+
+    @staticmethod
+    def other_type_check(other: Union[Distance, float, int]) -> bool:
+        if not isinstance(other, (Distance, float, int)):
+            raise TypeError("'other' is not 'Distance', 'float', 'int'!!!")
+        return isinstance(other, Distance)
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Union[Distance, float, int]) -> Distance:
+        return (
+            Distance(self.km + other.km)
+            if self.other_type_check(other) else
+            Distance(self.km + other)
+        )
+
+    def __iadd__(self, other: Union[Distance, int, float]) -> Distance:
+        if self.other_type_check(other):
+            self.km = self.km + other.km
+        elif isinstance(other, (float, int)):
+            self.km = self.km + other
+
+        return self
+
+    def __mul__(self, other: Union[float, int]) -> Distance:
+        if not self.other_type_check(other):
+            return Distance(self.km * other)
```

Comment:
The `__mul__` and `__truediv__` methods do not handle the case when `other` is an instance of `Distance`. According to the current implementation, if `other` is an instance of `Distance`, these methods will return `None`. This could lead to unexpected behavior. If the intention is to only allow multiplication and division by `float` or `int`, it would be better to raise a `TypeError` when `other` is a `Distance`.